lua/bson_lua_wrapper.c

```diff
@@ -0,0 +1,372 @@
+#include <stdlib.h>
+#include <string.h>
+ 
+#include <lua5.2/lua.h>
+#include <lua5.2/lauxlib.h>
+ 
+#include <bson_object.h>
+
+static int table_to_bson_object(lua_State *L, BsonObject *obj, char *errorMessage);
+static int table_to_bson_array(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int bson_object_to_table(lua_State *L, BsonObject *obj, char *errorMessage);
+static int bson_element_to_table(lua_State *L, BsonElement *element, char *errorMessage);
+static int bson_array_to_table(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int lbson_object_to_bytes(lua_State *L) {
+    BsonObject bson;
+    bson_object_initialize_default(&bson);
+    if (!lua_istable(L, 1)) {
+        return luaL_argerror(L, 1, "Arg 1 must be a table");
+    }
+
+    char errorMessage[255];
+    int success = table_to_bson_object(L, &bson, errorMessage);
+    if (success != 0) {
+        bson_object_deinitialize(&bson);
+        return luaL_argerror(L, 1, errorMessage);
+    }
+
+    uint8_t *bytes = bson_object_to_bytes(&bson);
+    if (bytes == NULL) {
+        bson_object_deinitialize(&bson);
+        return luaL_argerror(L, 1, "Map could not be parsed into bson");
+    }
+
+    size_t obj_size = bson_object_size(&bson);
+    bson_object_deinitialize(&bson);
+
+    char *byte_string = malloc(obj_size * sizeof(char));
+    size_t index = 0;
+    for (index = 0; index < obj_size; index++) {
+        byte_string[index] = bytes[index] & 0xFF;
+    }
+    free(bytes);
+
+    lua_pushlstring(L, (const char *)byte_string, obj_size);
+    free(byte_string);
+
+    return 1;
```

Comment:
@jacobkeeler `1` is success? Some documentation of functions would be rely good.

Reply:
The returned value is the number of items pushed to the stack (just how the Lua->C interface works). I can add documentation for all of this.